### PR TITLE
Update ndarray.py

### LINKED ISSUE #16333: std

### PR DESCRIPTION
**Pull request by PushpamJha14**
Close #16333

### CODE REVIEW

**Comment by soma2000-lang** [comment]
Hi @PushpamJha14,It seems you have missed out on the corresponding test for the function you have implemented. Is this PR a work in progress?

**Comment by PushpamJha14** [comment]
Hi @soma2000-lang ,
Work on this PR is complete, but I am getting a message of "10 workflows awaiting approval", what to do?

**Comment by soma2000-lang** [comment]
You newd to write tests also.

**Comment by PushpamJha14** [comment]
Do you mean that I have to create test cases for this

**Comment by soma2000-lang** [comment]
Write the test for the function you are adding

**Comment by PushpamJha14** [comment]
I have written test for the function, please check.

**Comment by soma2000-lang** [comment]
@PushpamJha14 pleawe run the tests locally first.

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by soma2000-lang** [comment]
@PushpamJha14 Can you please resolve conflicts

**Comment by PushpamJha14** [comment]
I have resolved the conflicts and updated the code, and
Screenshot (746)
 here i am attaching the results of local test.

**Comment by soma2000-lang** [comment]
Hi @PushpamJha14 Can you please look into the failing tests of the CI?

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by PushpamJha14** [comment]
Hi, @soma2000-lang I am not getting what to do to solve these failing test cases can you please help me.

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey would you update your branch with latest master branch for us to be able to run tests🙂

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 1]
```diff
@@ -3564,3 +3564,42 @@ def test_numpy___invert__(
...
+        available_dtypes=helpers.get_dtypes("float"),
+        valid_axis=True,
+    ),
+)
+def test_numpy_ndarray_std(
+    dtype_x_axis,
+    frontend_method_data,
+    init_flags,
+    method_flags,
+    frontend,
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:3583]:
```suggestion
    frontend,
    backend_fw,
```

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 2]
```diff
@@ -3564,3 +3564,42 @@ def test_numpy___invert__(
...
+        method_input_dtypes=input_dtypes,
+        init_all_as_kwargs_np={
+            "object": x[0],
+        },
+        method_all_as_kwargs_np={
+            "axis": axis,
+            "dtype": "float64",
+            "out": None,
+        },
+        frontend=frontend,
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:3598]:
```suggestion
        frontend=frontend,
        backend_to_test=backend_fw,
```

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey some fixtures are missing in the unit tests which is causing the failures.
I suggested them here in the comments🙂

After adding them you'll be able to debug the tests😊

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 3]
```diff
@@ -3564,3 +3564,44 @@ def test_numpy___invert__(
         method_flags=method_flags,
         on_device=on_device,
     )
+    
+# std
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="numpy.array",
+    method_name="std",
+    dtype_x_axis=helpers.dtype_values_axis(
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:3573]: There are assertion errors being thrown due to generated examples, given that the statistical functions have their specific strategy that avoids the overflow errors you can use that🙂
https://github.com/unifyai/ivy/blob/291d9277f4c57a3f3f9e607f5ac4266774f5dcf8/ivy_tests/test_ivy/test_functional/test_core/test_statistical.py#L12

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @PushpamJha14 

Just a few changes to get your PR merged😊
Check the comment below

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py (hunk 3 above)
Comment (PushpamJha14) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:3577]: Hi @hirwa-nshuti,
I have tried to made some changes in code but they are failing on more tests, so I tried to revert back previous code which has 2 failing test cases but now for that same is also failing on 56 test cases. I don't know what is wrong with this code could u please help me.

Comment (PushpamJha14) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:3577]: I messed up with the whole code, sorry for that. But to make it correct I discarded all commits and made new commits, these are the same codes that I was using just before you asked me to remove assertion error and at that time many test cases were passing but now more that half of them are failing, I have tried from my side to fix these but as I already said that I ended up with messing all the code.
So, I request you to please guide me.

**Pull request closed without merging**